notifications/Upgrading-OCP49-API.md

```diff
@@ -0,0 +1,73 @@
+
+Red Hat Openshift bases itself off of Kubernetes and then adds additional technologies to create the product we all use today. This means that for each version of Openshift, it is matched against a specific version of Kubernetes.
+
+For example, The version of Kubernetes that is used when we upgraded to Openshift 4.8 is Kubernetes version 1.21. We know this in advance since Red Hat in their release notes will put this version change details in the notes, since many technologies out there might not necessarily have compatibility details regarding Openshift versions, but do have this information for Kubernetes versions. Click [here](https://docs.openshift.com/container-platform/4.8/release_notes/ocp-4-8-release-notes.html#ocp-4-8-about-this-release) to see this for Openshift 4.8, which runs with Kubernetes version 1.21.
+
+## API Versioning
+
+<https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups-and-versioning>
+
+When we create or update Kubernetes resources (ie: Deployments, Routes, RoleBindings) inside Openshift, these resources will be defined with various pieces of information, including an API version. This information is processed and handled by the API Server accordingly.
+
+As an API resource matures over time in terms of how it is implemented and maintained, the API maintainers will choose to update the API version to match its level of maturity. For the current wave of API upgrades that will take place when we upgrade Openshift from 4.8 to 4.9 (thus upgrade from Kubernetes 1.21 to 1.22), the affected API resources will be all easily identifiable as being an API resource with version `v1beta1` and that version in Openshift 4.9 will need to be just `v1`.
+
+At present Openshift 4.8 will respond to the affected API resources both with API versions `v1beta1` and `v1`, but when we upgrade our Openshift clusters to 4.9, then using API version `v1beta1` for some specific API resources will no longer work and not be created or updated depending on the requests involved.
+
+
+## Which API Resources are Affected
+
+<https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22>
+
+There are a number of Kubernetes resources affected by this upcoming upgrade, but most of them do not apply to people performing development work for applications hosted inside Openshift. The applicable resources can be broken down into three specific Kubernetes resources.
+
+
+### Ingress
+<https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122>
+
+- API groups `networking.k8s.io` and `extensions` are to be replaced with just `networking.k8s.io`.
+- API version changes from `v1beta1` to `v1`.
+- Above link details changes regarding how the Ingress contents are defined.
+
+### Role and RoleBinding
+<https://kubernetes.io/docs/reference/using-api/deprecation-guide/#rbac-resources-v122>
+
+- API version changes from `v1beta1` to `v1`.
+- no other changes to note.
+
```

Comment:
Seems like we should include CustomResourceDefinition in this list, as v1beta1 will have to be replaced with v1.
https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122

Reply:
Product teams don't have the permissions to create CRDs/ Only cluster-admins can. All the operators AdvSol manages are already updated, or will update as part of the cluster upgrade.

Are the Platform Services owned operators updated? If not, we'll find out during the lab testing and roll out a CCM fix as part of the upgrade. Should still be transparent to Product Teams.

Reply:
There are kustomization.yaml files that reference a v1beta1 API version, but that is consistent with current Kubernetes docs and not listed in the Kubernetes API upgrade list.
`apiVersion: kustomize.config.k8s.io/v1beta1`
https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/